packages/manager/apps/dedicated/client/app/ip/components/list/firewall/game/ip-ip-firewall-game.controller.js

```diff
@@ -294,6 +294,47 @@ export default /* @ngInject */ function IpGameFirewallCtrl(
     });
   }
 
+  function checkPortsNotUsed() {
+    let isValid = true;
+    self.table.rules.forEach((rule) => {
+      if (rule.ports.from === rule.ports.to) {
+        if (self.rule.ports.from === self.rule.ports.to) {
+          if (self.rule.ports.from === rule.ports.from) {
+            isValid = false;
+          }
+        } else if (
+          self.rule.ports.from < self.rule.ports.to &&
+          rule.ports.from > self.rule.ports.from
+        ) {
+          isValid = false;
+        }
+      } else if (self.rule.ports.from === self.rule.ports.to) {
+        if (
+          self.rule.ports.from < rule.ports.to &&
+          self.rule.ports.from > rule.ports.from
+        ) {
+          isValid = false;
+        }
+      } else if (
+        (self.rule.ports.from > rule.ports.from &&
+          self.rule.ports.to < rule.ports.to) ||
+        (self.rule.ports.from < rule.ports.from &&
+          self.rule.ports.form < rule.ports.to &&
+          self.rule.ports.to < rule.ports.to) ||
+        (self.rule.ports.from > rule.ports.from &&
+          self.rule.ports.from < rule.ports.to &&
+          self.rule.ports.to > rule.ports.to) ||
+        (self.rule.ports.from < rule.ports.from &&
+          self.rule.ports.to > rule.ports.to) ||
+        (self.rule.ports.from < rule.ports.from &&
+          self.rule.ports.to > rule.ports.from)
+      ) {
+        isValid = false;
+      }
+    });
+    return isValid;
```

Comment:
```suggestion
   return !self.table.rules.some((rule) => {
      if (rule.ports.from === rule.ports.to) {
        if (self.rule.ports.from === self.rule.ports.to) {
          if (self.rule.ports.from === rule.ports.from) {
            return true;
          }
        } else if (
          self.rule.ports.from < self.rule.ports.to &&
          rule.ports.from > self.rule.ports.from
        ) {
          return true;
        }
      } else if (self.rule.ports.from === self.rule.ports.to) {
        if (
          self.rule.ports.from < rule.ports.to &&
          self.rule.ports.from > rule.ports.from
        ) {
          return true;
        }
      } else if (
        (self.rule.ports.from > rule.ports.from &&
          self.rule.ports.to < rule.ports.to) ||
        (self.rule.ports.from < rule.ports.from &&
          self.rule.ports.form < rule.ports.to &&
          self.rule.ports.to < rule.ports.to) ||
        (self.rule.ports.from > rule.ports.from &&
          self.rule.ports.from < rule.ports.to &&
          self.rule.ports.to > rule.ports.to) ||
        (self.rule.ports.from < rule.ports.from &&
          self.rule.ports.to > rule.ports.to) ||
        (self.rule.ports.from < rule.ports.from &&
          self.rule.ports.to > rule.ports.from)
      ) {
        return true;
      }
    });
```
This would be more performent (in some case) as it will stop looping at the first invalid rule

Reply:
You should also split it in sub functions with explicit names to ease the comprehension of the algorithm. Quite difficult to understand currently